mavros/src/lib/uas_frame_conversions.cpp

```diff
@@ -18,26 +19,135 @@
+// gives the ENU frame.  Similarly, a +PI rotation about X (East) followed by
+// a +PI/2 roation about Z (Up) gives the NED frame.
+static const Eigen::Quaterniond NED_ENU_Q = UAS::quaternion_from_rpy(M_PI, 0.0, M_PI_2);
+
+// Static quaternion needed for rotating between aircraft and base_link frames
+// +PI rotation around X (Forward) axis transforms from Forward, Right, Down (aircraft)
+// Fto Forward, Left, Up (base_link) frames.
+static const Eigen::Quaterniond AIRCRAFT_BASELINK_Q = UAS::quaternion_from_rpy(M_PI, 0.0, 0.0);
+
+static const Eigen::Affine3d NED_ENU_AFFINE(NED_ENU_Q);
+static const Eigen::Affine3d AIRCRAFT_BASELINK_AFFINE(AIRCRAFT_BASELINK_Q);
+
+static inline const Eigen::Affine3d &get_static_transform_affine(const UAS::STATIC_TRANSFORM transform)
+{
+	switch (transform) {
+	case UAS::STATIC_TRANSFORM::NED_TO_ENU:
+	case UAS::STATIC_TRANSFORM::ENU_TO_NED: {
+		return NED_ENU_AFFINE;
+		break;
+	}
+	case UAS::STATIC_TRANSFORM::AIRCRAFT_TO_BASELINK:
+	case UAS::STATIC_TRANSFORM::BASELINK_TO_AIRCRAFT: {
+		return AIRCRAFT_BASELINK_AFFINE;
+		break;
+	}
+	default: {
+		ROS_ASSERT(false);
+	}
+	}
+}
 
-//! +PI rotation around X (Roll) axis give us ROS or FCU representation
-static const Eigen::Quaterniond FRAME_ROTATE_Q = UAS::quaternion_from_rpy(M_PI, 0.0, 0.0);
 
 //! Transform for vector3
-static const Eigen::Transform<double, 3, Eigen::Affine> FRAME_TRANSFORM_VECTOR3(FRAME_ROTATE_Q);
+//static const Eigen::Transform<double, 3, Eigen::Affine> FRAME_TRANSFORM_VECTOR3(FRAME_ROTATE_Q);
+
+
+Eigen::Quaterniond UAS::transform_orientation(const Eigen::Quaterniond &q, const UAS::STATIC_TRANSFORM transform)
+{
+	//Transform the attitude representation from frame to frame.  The proof for this transform can
+	//be seen http://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/transforms/
+	switch (transform) {
+	case STATIC_TRANSFORM::NED_TO_ENU:
+	case STATIC_TRANSFORM::ENU_TO_NED: {
+		return NED_ENU_Q * q;
+		break;
+	}
+	case STATIC_TRANSFORM::AIRCRAFT_TO_BASELINK:
+	case STATIC_TRANSFORM::BASELINK_TO_AIRCRAFT: {
+		return q * AIRCRAFT_BASELINK_Q;
+		break;
+	}
+	default: {
+		//We don't know how to express the attitude WRT an undefined
+		//frame.
+		ROS_ASSERT(false);
+		return q;
+		break;
+	}
+	}
+}
 
+Eigen::Vector3d UAS::transform_static_frame(const Eigen::Vector3d &vec, const UAS::STATIC_TRANSFORM transform)
+{
+	auto affine_transform = get_static_transform_affine(transform);
+	return affine_transform * vec;
+}
 
-Eigen::Quaterniond UAS::transform_frame(const Eigen::Quaterniond &q)
+UAS::Covariance3d UAS::transform_static_frame(const Covariance3d &cov, const UAS::STATIC_TRANSFORM transform)
 {
-	return FRAME_ROTATE_Q * q * FRAME_ROTATE_Q.inverse();
+	switch (transform) {
+	case STATIC_TRANSFORM::NED_TO_ENU:
+	case STATIC_TRANSFORM::ENU_TO_NED: {
+		Covariance3d cov_out_;
+		EigenMapConstCovariance3d cov_in(cov.data());
+		EigenMapCovariance3d cov_out(cov_out_.data());
```

Comment:
Move map preparation out of switch - duplication of code.
Also i'm not sure that `{}` are necessary in that switches.